frontend/src/components/FacilityBooking/ViewBookingCard.tsx

```diff
@@ -0,0 +1,182 @@
+  flex-direction: row;
+  justify-content: center;
+  width: 100%;
+  height: 100%;
+  top: 0;
+  left: 0;
+  right: 0;
+  bottom: 0;
+  background-color: rgba(0, 0, 0, 0.7);
+  z-index: 202;
+  cursor: pointer;
+`
+
+const BookingContainer = styled.div`
+  position: absolute;
+  padding-bottom: 20px;
+  top: 80px;
+  gap: 10px;
+  width: 85%;
+  max-width: 400px;
+  cursor: pointer;
+  background-color: #ffffff;
+  border-radius: 20px;
+  box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
+  display: flex;
+  flex-direction: column;
+  justify-content: space-evenly;
+  align-items: center;
+  z-index: 205;
+`
+
+const BookingHeader = styled.p`
+  width: 100%;
+  height: 20%;
+  cursor: pointer;
+  background-color: #f1f3f7;
+  border-radius: 20px 20px 0px 0px;
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  z-index: 2;
+  text-align: center;
+  padding: 20px 0px;
+`
+
+const TelegramDetailsContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+`
+
+const EventDetailsContainer = styled.div`
+  display: flex;
+  flex-direction: row;
+`
+
+const EventTimingContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+`
+
+const ExitButtonContainer = styled.img`
+  width: 50px;
+  height: 50px;
+  cursor: pointer;
+`
+
+const TelegramButtonContainer = styled.img`
+  width: 39px;
+  height: 39px;
+  cursor: pointer;
+`
+
+const StyledText = styled.text<{ fontSize?: string; fontWeight?: string; color?: string }>`
+  font-size: ${(props) => props.fontSize ?? `14px`};
+  color: ${(props) => props.color ?? `#191919`};
+  font-weight: ${(props) => props.fontWeight ?? `400`};
+  text-align: center;
+  line-height: 22px;
+`
+
+type Props = {
+  booking?: Booking | null
+  onClickFunction: Dispatch<SetStateAction<boolean | undefined>>
+}
+
+export const ViewBookingCard = (props: Props) => {
+  const [telegramHandle, setTelegramHandle] = useState<string>()
+  const [bookingStartTimeUnix, setBookingStartTimeUnix] = useState<number>(0)
+  const [bookingEndTimeUnix, setBookingEndTimeUnix] = useState<number>(0)
+
+  useEffect(() => {
+    fetchTelegram(props.booking?.userID)
+    setBookingStartTimeUnix(props.booking?.startTime == undefined ? 0 : props.booking?.startTime)
+    setBookingEndTimeUnix(props.booking?.endTime == undefined ? 0 : props.booking?.endTime)
+  })
+
+  const ExitButton = () => {
+    return <ExitButtonContainer src={ViewBookingCardButton} onClick={() => props.onClickFunction(false)} />
+  }
+
+  const TelegramButton = () => {
+    return <TelegramButtonContainer src={ViewBookingCardUserIcon} />
+  }
+
+  const fetchTelegram = (userID: string | undefined) => {
+    if (userID) {
+      try {
+        fetch(DOMAIN_URL.FACILITY + ENDPOINTS.TELEGRAM_HANDLE + '/' + userID, {
+          method: 'GET',
+          mode: 'cors',
+        })
+          .then((resp) => resp.json())
+          .then((data) => {
+            if (data.data === '' || data.data === undefined) {
```

Comment:
can use `!data.data` instead? then if data is empty or undefined then just set empty string as tele handle, no need to catch error?

Reply:
Catching error was meant for fetch function